aten/src/ATen/native/cuda/AdaptiveMaxPooling2d.cu

```diff
@@ -0,0 +1,472 @@
+#include "ATen/ATen.h"
+#include "ATen/cuda/CUDAApplyUtils.cuh"
+#include "ATen/cuda/CUDAContext.h"
+#include "ATen/NativeFunctions.h"
+#include "ATen/TensorUtils.h"
+#include "ATen/Utils.h"
+#include "c10/util/Exception.h"
+#include <THC/THCGeneral.h>
+#include "THC/THCNumerics.cuh"
+
+#include <algorithm>
+#include <cfloat>
+#include <cmath>
+
+#define CUDA_MAX_THREADS 1024   // this is safe, in reality 256 is our limit
+
+#define START_IND(a,b,c) (int)std::floor((float)(a * c) / b)
+#define END_IND(a,b,c) (int)std::ceil((float)((a + 1) * c) / b)
```

Comment:
Macro here is suboptimal, but it doesn't have to get fixed in this PR.

Reply:
Oh, but I do see you fixed it in your other file ;)

Reply:
@ezyang I tried it now for the follow-up PR, but CUDA does not like it (apparently the function is not inlined):

```
error: calling a __host__ function("at::native::_NV_ANON_NAMESPACE::start_index") from a __global__ function("at::native::_NV_ANON_NAMESPACE::adaptivemaxpool<double> ") is not allowed
```

Reply:
Sorry for the noise, obviously it needs `__device__`. :)